### PR TITLE
Framed substream default maximum frame size

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
Added a maximum frame length constant for a framed substream, equal to the default Tokio setting, so it can be visible in the Tari code and not dependent on the default frame size.

## Motivation and Context
Difficult to find this default setting for tuning when it is hidden away in the Tokio libraries. This will also protect the Tari code base from future Tokio code changes.

## How Has This Been Tested?
Tested in a base node exe.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: comms/src/protocol/messaging/protocol.rs [hunk 1]
```diff
@@ -205,7 +206,9 @@ impl MessagingProtocol {
 
     pub fn framed<TSubstream>(socket: TSubstream) -> Framed<IoCompat<TSubstream>, LengthDelimitedCodec>
     where TSubstream: AsyncRead + AsyncWrite + Unpin {
-        Framed::new(IoCompat::new(socket), LengthDelimitedCodec::new())
+        let mut codec = LengthDelimitedCodec::new();
+        codec.set_max_frame_length(MAX_FRAME_LENGTH_MB * 1_024 * 1_024);
+        Framed::new(IoCompat::new(socket), codec)
```
Comment (sdbondi) [inline comms/src/protocol/messaging/protocol.rs]: nit: The builder is the idiomatic way to customize the codec

```suggestion
        let codec = LengthDelimitedCodec::builder()
        .max_frame_length(MAX_FRAME_LENGTH)
        .new_codec();
        Framed::new(IoCompat::new(socket), codec)
```

File: comms/src/protocol/messaging/protocol.rs [hunk 2]
```diff
@@ -61,6 +61,7 @@ const MAX_SEND_RETRIES: usize = 1;
 /// The maximum amount of inbound messages to accept within the `RATE_LIMIT_RESTOCK_INTERVAL` window
 const RATE_LIMIT_CAPACITY: usize = 10;
 const RATE_LIMIT_RESTOCK_INTERVAL: Duration = Duration::from_millis(100);
+const MAX_FRAME_LENGTH_MB: usize = 8;
```
Comment (sdbondi) [inline comms/src/protocol/messaging/protocol.rs]: nit: We may want to do sub-Mb adjustment 
```suggestion
const MAX_FRAME_LENGTH: usize = 8 * 1_024 * 1_024;
```

**Review by sdbondi — COMMENTED** [review]
Small nits - but otherwise looks good

File: comms/src/protocol/messaging/protocol.rs (hunk 1 above)
Comment (hansieodendaal) [reply comms/src/protocol/messaging/protocol.rs]: Committed, rebased and force pushed from local

File: comms/src/protocol/messaging/protocol.rs (hunk 2 above)
Comment (hansieodendaal) [reply comms/src/protocol/messaging/protocol.rs]: Committed, rebased and force pushed from local

**Review by sdbondi — APPROVED** [review]

**Pull request merged**